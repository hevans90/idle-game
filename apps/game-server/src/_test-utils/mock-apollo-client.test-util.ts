import {
  ApolloClient,
  ApolloQueryResult,
  InMemoryCache,
  NormalizedCacheObject,
} from '@apollo/client';
import { MockSubscriptionLink } from '@apollo/client/testing';

export const mockApolloClient = () => {
  const link = new MockSubscriptionLink();

  return {
    client: new ApolloClient({
      link,
      cache: new InMemoryCache(),
    }),
    link,
  };
};

export const mockQuery = <T>(
  mockClient: ApolloClient<NormalizedCacheObject>,
  data: T
) => {
  jest.spyOn(mockClient, 'query').mockImplementation(
    () =>
      new Promise<ApolloQueryResult<T>>((resolve, reject) => {
        resolve({
          data,
          loading: false,
          networkStatus: 1,
        });
      })
  );
};
