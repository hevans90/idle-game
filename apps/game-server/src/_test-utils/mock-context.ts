import { ApolloClient, NormalizedCacheObject } from '@apollo/client';
import { Auth0API } from '../datasources/auth0';
import { Context } from '../datasources/context';
import { HasuraAPI } from '../datasources/hasura-api';
import { HasuraEmpirePurchases } from '../datasources/hasura-empire-purchases';
import { HasuraEmpireResourceModifiers } from '../datasources/hasura-empire-resource-modifiers';
import { HasuraQuestProgression } from '../datasources/hasura-quest-progression';
import { MinioAPI } from '../datasources/minio';

export const mockContext = (
  mockClient: ApolloClient<NormalizedCacheObject>,
  userId: string
): Context => {
  return {
    id: userId,
    req: undefined,
    user: undefined,
    dataSources: {
      minio: new MinioAPI(null),
      hasuraAPI: new HasuraAPI(mockClient),
      hasuraEmpirePurchases: new HasuraEmpirePurchases(mockClient),
      hasuraQuestProgression: new HasuraQuestProgression(mockClient),
      hasuraEmpireResourceModifiers: new HasuraEmpireResourceModifiers(
        mockClient
      ),
      auth0API: new Auth0API(),
    },
  };
};
