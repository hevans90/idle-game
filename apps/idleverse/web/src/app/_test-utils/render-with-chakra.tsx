import { ChakraProvider } from '@chakra-ui/react';
import { render } from '@testing-library/react';
import React from 'react';
import { theme } from '../_theme/theme';

export const renderWithChakra = (ui) => {
  const Wrapper = ({ children }) => (
    <ChakraProvider theme={theme}>{children}</ChakraProvider>
  );

  return render(ui, { wrapper: Wrapper });
};
