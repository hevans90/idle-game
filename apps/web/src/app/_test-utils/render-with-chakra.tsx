import { ChakraProvider } from '@chakra-ui/react';
import { theme } from '@idleverse/theme';
import { render } from '@testing-library/react';

export const renderWithChakra = (ui) => {
  const Wrapper = ({ children }) => (
    <ChakraProvider theme={theme({ primary: 'gray', secondary: 'teal' })}>
      {children}
    </ChakraProvider>
  );

  return render(ui, { wrapper: Wrapper });
};
