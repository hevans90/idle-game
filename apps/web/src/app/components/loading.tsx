import { Box, Text, TypographyProps } from '@chakra-ui/react';

type LoadingProps = {
  height?: string;
  width?: string;
  text?: string;
  fontSize?: TypographyProps['fontSize'];
};

export const Loading = ({
  height = '100vh',
  width = '100vw',
  text = 'Loading...',
  fontSize = '4xl',
}: LoadingProps | undefined) => (
  <Box
    height={height}
    width={width}
    display="flex"
    alignItems="center"
    justifyContent="center"
  >
    <Text fontSize={fontSize}>{text}</Text>
  </Box>
);
