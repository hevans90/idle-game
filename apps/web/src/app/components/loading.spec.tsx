import { renderWithChakra } from '../_test-utils/render-with-chakra';
import { Loading } from './loading';

describe('loading', () => {
  it('should render', () => {
    const { container } = renderWithChakra(<Loading></Loading>);
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('should render custom text', () => {
    const { container } = renderWithChakra(
      <Loading text="Loading memes"></Loading>
    );
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('should render custom text and size', () => {
    const { container } = renderWithChakra(
      <Loading width="50%" height="100%" text="Loading memes"></Loading>
    );
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
