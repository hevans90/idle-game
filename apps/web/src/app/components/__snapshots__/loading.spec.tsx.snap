// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading should render 1`] = `
<div>
  <div
    class="css-sidcim"
  >
    <p
      class="chakra-text css-17vwc26"
    >
      Loading
    </p>
    <span
      class="chakra-text css-a0yd5z"
    >
      .
    </span>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;

exports[`loading should render custom text 1`] = `
<div>
  <div
    class="css-sidcim"
  >
    <p
      class="chakra-text css-17vwc26"
    >
      Loading memes
    </p>
    <span
      class="chakra-text css-a0yd5z"
    >
      .
    </span>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;

exports[`loading should render custom text and size 1`] = `
<div>
  <div
    class="css-n4refw"
  >
    <p
      class="chakra-text css-17vwc26"
    >
      Loading memes
    </p>
    <span
      class="chakra-text css-a0yd5z"
    >
      .
    </span>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
